corehq/ex-submodules/pillowtop/reindexer/reindexer.py

```diff
@@ -111,26 +117,54 @@ def process_bulk_docs(self, docs):
         if len(docs) == 0:
             return True
 
+        pillow_logging.info("Processing batch of %s docs", len((docs)))
+
         changes = [self._doc_to_change(doc) for doc in docs]
         error_collector = ErrorCollector()
 
         bulk_changes = build_bulk_payload(self.index_info, changes, self.doc_transform, error_collector)
 
+        for change, exception in error_collector.errors:
+            pillow_logging.error("Error procesing doc %s: %s", change.id, exception)
+
         max_payload_size = pow(10, 8)  # ~ 100Mb
         payloads = prepare_bulk_payloads(bulk_changes, max_payload_size)
         if len(payloads) > 1:
-            pillow_logging.info("payload split into %s parts" % len(payloads))
-
-        pillow_logging.info("sending payload,%s" % len(changes))
+            pillow_logging.info("Payload split into %s parts" % len(payloads))
 
         for payload in payloads:
-            self.es.bulk(payload)
-
-        for change, exception in error_collector.errors:
-            pillow_logging.error("Unable to process change %s: %s", change.id, exception)
+            success = self._send_payload_with_retries(payload)
+            if not success:
+                # stop the reindexer if we're unable to send a payload to ES
+                return False
 
         return True
 
+    def _send_payload_with_retries(self, payload):
```

Comment:
yea, I tested this in india and got a few timeout errors which required restarting the reindexer